apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx

```diff
@@ -509,48 +520,51 @@ var WorkshopForm = React.createClass({
           <br/>
           <Row>
             <Col sm={4}>
-              <Input
-                type="text"
-                label="Location Name"
-                value={this.state.location_name || ''}
-                onChange={(event) => {this.handleFieldChange('location_name', event.target.value);}}
-                bsStyle={validation.style.location_name}
-                help={validation.help.location_name}
-                maxLength={255}
-                style={this.props.readOnly && styles.readOnlyInput}
-                disabled={this.props.readOnly}
-              />
+              <FormGroup validationState={validation.style.location_name}>
+                <ControlLabel>Location Name</ControlLabel>
+                <FormControl
+                  type="text"
+                  value={this.state.location_name || ''}
+                  onChange={(event) => {this.handleFieldChange('location_name', event.target.value);}}
+                  maxLength={255}
+                  style={this.props.readOnly && styles.readOnlyInput}
+                  disabled={this.props.readOnly}
+                />
+                <HelpBlock>{validation.help.location_name}</HelpBlock>
+              </FormGroup>
             </Col>
             <Col sm={6}>
-              <Input
-                type="text"
-                key={this.state.useAutocomplete} // Change key to force re-draw
-                className="location-autocomplete"
-                label="Location Address"
-                value={this.state.location_address || ''}
-                placeholder="Enter a location"
-                onChange={(event) => {this.handleFieldChange('location_address', event.target.value);}}
-                bsStyle={validation.style.location_address}
-                help={validation.help.location_address}
-                maxLength={255}
-                style={this.props.readOnly && styles.readOnlyInput}
-                disabled={this.props.readOnly}
-              />
+              <FormGroup validationState={validation.style.location_address}>
+                <ControlLabel>Location Address</ControlLabel>
+                <FormControl
+                  type="text"
+                  key={this.state.useAutocomplete} // Change key to force re-draw
+                  ref={(ref) => this.locationAddressControl = ReactDOM.findDOMNode(ref)}
+                  value={this.state.location_address || ''}
+                  placeholder="Enter a location"
+                  onChange={(event) => {this.handleFieldChange('location_address', event.target.value);}}
```

Comment:
here (and a bunch of places in this PR) you're creating a function inside of the render method, which is something we should avoid when we can. This was the case before your change too, but I think we'd be better off doing one of two things
(1) having a method `this.handleLocationAddressFieldChange` (i.e. one instance method per possible field change)
(2) In `handleFieldChange` look at event.target and possibly derive the changed field from it?

This might be a little bit of a micro-optimization, but the reasoning is that render methods have the potential to be called many times, and when you create a function within it, that function might be created many times. When the function is instead a class member, it will only be created once instead of once per render.

Up to you if you change this or not in this PR